File: go/150.go
package main

import "strconv"

func evalRPN(tokens []string) int {
	var st []int = make([]int, 0)
	for _, v := range tokens {
		switch v {
		case "+":
			st[len(st)-2] += st[len(st)-1]
			st = st[:len(st)-1]
		case "-":
			st[len(st)-2] -= st[len(st)-1]
			st = st[:len(st)-1]
		case "*":
			st[len(st)-2] *= st[len(st)-1]
			st = st[:len(st)-1]
		case "/":
			st[len(st)-2] /= st[len(st)-1]
			st = st[:len(st)-1]
		default:
			val, _ := strconv.Atoi(v)
			st = append(st, val)

		}
	}
	return st[len(st)-1]
}

File: python/150.py
from typing import List


class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        st = []
        for s in tokens:
            if s == '+':
                st[-2] += st[-1]
                st = st[:-1]
            elif s == '-':
                st[-2] -= st[-1]
                st = st[:-1]
            elif s == '*':
                st[-2] *= st[-1]
                st = st[:-1]
            elif s == '/':
                st[-2] = st[-2] // st[-1]
                st = st[:-1]
            else:
                st.append(int(s))

        return st[0]
